inference-gateway/src/events_reporter.rs

```diff
@@ -0,0 +1,303 @@
+        AND completed_at <= $2
+    GROUP BY
+        organization_id,
+        instance_id,
+        model
+        "#,
+        start_time,
+        end_time
+    )
+    .fetch_all(dbclient)
+    .await?;
+
+    Ok(rows_to_events(rows))
+}
+
+fn rows_to_events(rows: Vec<UsageData>) -> Vec<Events> {
+    rows.into_iter()
+        .map(|row| {
+            // Parse the completed_at string into a DateTime<Utc> and convert to hour
+            let completed_at = row
+                .completed_at
+                .unwrap()
+                .with_timezone(&Utc)
+                .format("%Y%m%d%H")
+                .to_string();
+
+            Events {
+                idempotency_key: format!("{}-{}-{}", row.instance_id, row.model, completed_at),
+                organization_id: row.organization_id,
+                instance_id: row.instance_id,
+                payload: Payload {
+                    completed_at: row.completed_at.unwrap_or_default().to_string(),
+                    model: row.model,
+                    prompt_tokens: row.prompt_tokens.unwrap_or(0).to_string(),
+                    completion_tokens: row.completion_tokens.unwrap_or(0).to_string(),
+                },
+            }
+        })
+        .collect()
+}
+
+async fn get_reporter_watermark(conn: &PgPool) -> Result<Option<ReporterWatermark>> {
+    sqlx::query_as!(
+        ReporterWatermark,
+        "SELECT last_reported_at FROM billing.reporter_watermark LIMIT 1"
+    )
+    .fetch_optional(conn)
+    .await
+    .map_err(Into::into)
+}
+
+fn start_of_the_hour(datetime: DateTime<Utc>) -> DateTime<Utc> {
+    // Safe unwrap since, according to chrono docs, Utc will never have double mappings
+    Utc.with_ymd_and_hms(
+        datetime.year(),
+        datetime.month(),
+        datetime.day(),
+        datetime.hour(),
+        0,
+        0,
+    )
+    .unwrap()
+}
+
+fn get_hourly_chunks(
+    last_reported_at: DateTime<Utc>,
+    now: DateTime<Utc>,
+) -> Vec<(DateTime<Utc>, DateTime<Utc>)> {
+    let last_complete_hour = start_of_the_hour(now);
+
+    let mut chunks = Vec::new();
+    let mut chunk_start = start_of_the_hour(last_reported_at);
+
+    while chunk_start < last_complete_hour {
+        let chunk_end = chunk_start + ChronoDuration::hours(1) - ChronoDuration::nanoseconds(1);
+        chunks.push((chunk_start, chunk_end));
+        chunk_start = chunk_start + ChronoDuration::hours(1);
+    }
+
+    chunks
+}
+
+pub async fn events_reporter() -> Result<()> {
+    // Run once per hour
+    const SYNC_PERIOD: Duration = Duration::from_secs(60 * 60);
+
+    let pg_conn_url = env::var("POSTGRES_QUEUE_CONNECTION")
+        .with_context(|| "POSTGRES_QUEUE_CONNECTION must be set")?;
+    let dbclient = connect(&pg_conn_url, 5).await?;
+
+    let queue = PGMQueueExt::new(pg_conn_url, 5).await?;
+
+    // TODO: Need to set this env variable
+    let metrics_events_queue =
+        env::var("BILLING_EVENTS_QUEUE").expect("BILLING_EVENTS_QUEUE must be set");
```

Comment:
I think the first one is supposed to be the tembo AI DB so the env name and variable name seems off, and also I think it should not have a queue associated to it. The second one the variable name seems off. If I understand correctly, this should connect to the AI db and the billing queue only